modules/lib/core/index.d.ts

```diff
@@ -140,6 +135,27 @@ export interface Content<
     variantOf?: string;
 }
 
+export interface FragmentContent extends CommonContent {
+    fragment: LayoutComponent | PartComponent
+    type: 'portal:fragment'
+}
+
+export interface PageContent<
+    Data = Record<string, unknown>,
+    Type extends string = string,
+    Page extends Component = Component,
+> extends CommonContent {
+    data: Data;
+    page: Page;
+    type: Type;
+}
+
+export type Content<
+    Data = Record<string, unknown>,
+    Type extends string = string,
+    Page extends Component = Component,
+> = FragmentContent | PageContent;
```

Comment:
## Problematic union

I think this union will be problematic. I think it will force us to check `content.type !== "portal:fragment"` every time we use `Content`.

How about you just build the fragment support into the existing `Content`?

```diff
export interface Content<
    Data = Record<string, unknown>,
    Type extends string = string,
    Page extends Component = Component,
> {
    _id: string;
    _name: string;
    _path: string;
    _score?: number;
    creator: UserKey;
    modifier?: UserKey;
    createdTime: string;
    modifiedTime?: string;
    owner: string;
    data: Data;
    type: Type;
    displayName: string;
    hasChildren: boolean;
    language?: string;
    valid: boolean;
    originProject?: string;
    childOrder?: string;
    _sort?: object[];
    page: Page;
    x: XpXData;
    attachments: Record<string, Attachment>;
    publish?: PublishInfo;
    workflow?: {
        state: 'IN_PROGRESS' | 'PENDING_APPROVAL' | 'REJECTED' | 'READY';
        checks?: Record<string, 'PENDING' | 'REJECTED' | 'APPROVED'>;
    };
    inherit?: ('CONTENT' | 'PARENT' | 'NAME' | 'SORT')[];
    variantOf?: string;
+   fragment: Type extends "portal:fragment" 
+      ? LayoutComponent | PartComponent
+      : undefined;
}
```

If someone needs to handle fragments in their code, they can do this:

```typescript
type ContentFragment = Content<{}, "portal:fragment">; // This should have `fragment` on it.
```

----

## Problematic complexity

I think the type `Content` is the most important TypeScript-type in all of XP. It is imperative that developers can understand it and work with it easily.

I would be very careful to make it more complex (like a union).